app/src/main/java/com/anysoftkeyboard/ime/AnySoftKeyboardSuggestions.java

```diff
@@ -458,14 +455,6 @@ public void onUpdateSelection(
             mUndoCommitCursorPosition = newSelStart;
         }
 
-        final boolean isExpectedEvent = SystemClock.uptimeMillis() < mExpectingSelectionUpdateBy;
```

Comment:
i'm not sure I understand the problem.

Reply:
If you don't understand the problem, there was probably none to begin with. I think you added that line long ago.

I don't know why you needed to discard "expected events", I suppose it was just a heuristic to make the keyboard run quicker, and it went wrong.

Reply:
I think I did that since I assumed that if the event has not been delivered after 1500ms, it will never arrive....